From 52727146db11d3f7f454e15865655d03496a3300 Mon Sep 17 00:00:00 2001
From: Ulf Hermann <ulf.hermann@qt.io>
Date: Tue, 8 Jan 2019 15:54:40 +0100
Subject: [PATCH] Fix build with -c++std=c++11

The compile test for make_unique needs the C++ standard as argument in
order to provide meaningful results. For C++ >= 14, however, we don't
have to run the test at all as we are guaranteed to have make_unique.
Therefore, we specialize the test for C++11.

Change-Id: If13b1f825224a1e10841263c4de57d1b14cce650
Fixes: QTBUG-72904
Reviewed-by: Lars Knoll <lars.knoll@qt.io>
---
 src/qml/configure.json | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/qml/configure.json b/src/qml/configure.json
index 878ec0119..17d1f800a 100644
--- a/src/qml/configure.json
+++ b/src/qml/configure.json
@@ -20,7 +20,8 @@
                 "include": "memory",
                 "main": [
                     "std::unique_ptr<int> ptr = std::make_unique<int>();"
-                ]
+                ],
+                "qmake": "CONFIG += c++11"
             }
         }
     },
@@ -28,7 +29,7 @@
     "features": {
         "cxx14_make_unique": {
             "label": "C++14 make_unique",
-            "condition": "tests.cxx14_make_unique",
+            "condition": "features.c++14 || tests.cxx14_make_unique",
             "output": [ "privateFeature" ]
         },
         "qml-network": {
-- 
2.20.1.windows.1

